### PR TITLE
fix static check failures in component-base/metrics

### PR DESCRIPTION
**Pull request by Zhuzhenghao**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:

/kind cleanup

**What this PR does / why we need it**:
```
Errors from staticcheck:

vendor/k8s.io/component-base/metrics
```

**Which issue(s) this PR fixes**:

Ref: #81657

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
NONE
```

### CODE REVIEW

**Review by draveness — CHANGES_REQUESTED** [review]
You are supposed to remove this package from https://github.com/kubernetes/kubernetes/blame/master/hack/.staticcheck_failures

/ok-to-test

File: staging/src/k8s.io/component-base/metrics/metric.go [hunk 1]
```diff
@@ -165,7 +165,6 @@ func (c *selfCollector) Collect(ch chan<- prometheus.Metric) {
 // no-op vecs for convenience
 var noopCounterVec = &prometheus.CounterVec{}
 var noopHistogramVec = &prometheus.HistogramVec{}
-var noopSummaryVec = &prometheus.SummaryVec{}
```
[the remark this answers is not in the document]
Comment (Zhuzhenghao) [reply staging/src/k8s.io/component-base/metrics/metric.go]: OK，I wil ignore the linting error

[the remark this answers is not in the document]
Comment (Zhuzhenghao) [reply staging/src/k8s.io/component-base/metrics/metric.go]: Crying >.. 
Can u tell me what will it be used for?

**Comment by Zhuzhenghao** [comment]
> You are supposed to remove this package from https://github.com/kubernetes/kubernetes/blame/master/hack/.staticcheck_failures
> 
> /ok-to-test

Thx, I have removed that package

**Pull request closed without merging**